src/main/java/com/minecolonies/coremod/commands/DeleteColonyCommand.java

```diff
@@ -45,46 +49,41 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
     {
-        int colonyId = -1;
-        UUID mayorID = null;
-
-        if (args.length != 0)
+        if(args.length == 0)
         {
-            try
-            {
-                colonyId = Integer.parseInt(args[0]);
-            }
-            catch (final NumberFormatException e)
-            {
-                mayorID = getUUIDFromName(sender, args);
-            }
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            return;
         }
 
-        final IColony colony;
-        if (colonyId == -1)
+        if (!canCommandSenderUseCommand(SHOWCOLONYINFO))
         {
-            colony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID);
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NOT_PERMITTED));
+            return;
         }
-        else
+
+        int colonyId = getIthArgument(args, 0, -1);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'colonyId' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)